site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5171,6 +5187,35 @@ public function getInstructorLevelAccessCourse(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    /**
+     * @return array<Course>
+     */
+    public function getSelfRegistrationCourses(string $user_id): array {
+        $query = <<<SQL
+SELECT c.*, t.name AS term_name FROM courses c, terms t
+WHERE c.self_registration_type > ? AND c.status = 1 and c.course NOT IN (
```

Comment:
using `>` feels breakable. Is there any reason not to use `!=`?

Reply:
0 is NO_SELF_REGISTRATION, any number is a different type of self registration, so we want to show all courses that are available to self registration, so greater than 0 is some type of self registration is allowed, so we want to display it.